include/networkit/graph/Graph.hpp

```diff
@@ -2162,6 +2186,57 @@ std::pair<count, count> Graph::removeAdjacentEdges(node u, Condition condition,
     return {removedEdges, removedSelfLoops};
 }
 
+template <class Lambda>
+void Graph::sortEdges(Lambda lambda) {
+    if (!isWeighted()) {
+        WARN("The graph is not weighted, sorting edges by node id.");
+        sortEdges();
+        return;
+    }
+
+    const auto sortAdjacencyArrays = [&](node u, std::vector<node> &adjList,
+                                         std::vector<edgeweight> &weights,
+                                         std::vector<edgeid> &edgeIds) -> void {
+        const auto deg = adjList.size();
+        std::vector<index> indices(deg);
+        std::iota(indices.begin(), indices.end(), 0);
+        std::sort(indices.begin(), indices.end(), [&](auto a, auto b) -> bool {
+            return lambda(WeightedEdge{u, adjList[a], weights[a]},
+                          WeightedEdge{u, adjList[b], weights[b]});
+        });
+
+        std::vector<node> sortedAdjList(deg);
+        std::vector<edgeweight> sortedWeights(deg);
+        for (index i = 0; i < deg; ++i) {
+            sortedAdjList[i] = adjList[indices[i]];
+            sortedWeights[i] = weights[indices[i]];
+        }
+
+        adjList = std::move(sortedAdjList);
+        weights = std::move(sortedWeights);
```

Comment:
Wouldn't it be possible to apply this permutation in-place using just a single temporary value? This could also be done in a utility method that is then applied to the vectors as needed.

Reply:
That's a good idea, I implemented this feature in `ArrayTools::applyPermutation`. Unfortunately I a have to use a bool vector to keep track of which elements have already been swapped, not sure if we can avoid that.

Reply:
It is not easily possible to avoid this, the best solution seems to use a bit of the permutation to mark the swapped elements (e.g., used signed integers and change each value `x` to `- 1 - x`). Afterwards, one would need to reverse this.

Reply:
This would work if the permutation does not use all the bits right? If it does, we still need to use an additional vector. I could add a condition that checks if the length of the permutation requires all the bits, and use the additional vector only if necessary.

Reply:
If the permutation needs all 64 bits, this means that the graph has more than 2^63 nodes. I do not think that NetworKit intends to support such graphs in the foreseeable future.

Reply:
So do we always assume that the permutation is a 64-bit integer? Since this is a general utility method it might make sense to support also shorter integers (e.g., (u)int8_t).